src/xlsx.js

```diff
@@ -0,0 +1,99 @@
+export class ExcelWorkbook {
+  constructor(workbook) {
+    Object.defineProperty(this, "_", {value: workbook});
+  }
+  sheetNames() {
+    return this._.worksheets.map((sheet) => sheet.name);
+  }
+  sheet(name, {range, headers = false} = {}) {
+    const sheet = this._.getWorksheet(
+      typeof name === "number" ? this.sheetNames()[name] : name + ""
+    );
+    if (!sheet) throw new Error(`Sheet not found: ${name}`);
+    return extract(sheet, {range, headers});
+  }
+}
+
+function extract(sheet, {range, headers}) {
+  let [[c0, r0], [c1, r1]] = parseRange(range, sheet);
+  const seen = new Set();
+  const names = [];
+  const headerRow = headers && sheet._rows[r0++];
+  function name(n) {
+    if (!names[n]) {
```

Comment:
```suggestion
    if (!(n in names)) {
```

This avoids the chance that the name could be a falsely value (e.g., the empty string).

Reply:
I don't think we'd want the empty string as a column name, it's a case where I'd rather replace it with the column’s AA value.

Reply:
hm, what about `"0"`?